src/DataViews/ModulesDataViewTest.cpp

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_format.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <string>
+
+#include "ClientData/ProcessData.h"
+#include "DataViews/DataView.h"
+#include "DataViews/ModulesDataView.h"
+#include "DisplayFormats/DisplayFormats.h"
+#include "MockAppInterface.h"
+#include "OrbitBase/File.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/TemporaryFile.h"
+#include "OrbitBase/TestUtils.h"
+#include "module.pb.h"
+
+using orbit_client_data::ModuleData;
+using orbit_client_data::ModuleInMemory;
+using orbit_grpc_protos::ModuleInfo;
+
+namespace {
+
+constexpr size_t kNumModules = 3;
+constexpr std::array<uint64_t, kNumModules> kStartAddresses{0x1000, 0x2000, 0x3000};
+constexpr std::array<uint64_t, kNumModules> kEndAddresses{0x1100, 0x2100, 0x3100};
+constexpr std::array<uint64_t, kNumModules> kSizes{100, 200, 300};
+const std::array<std::string, kNumModules> kNames{"module_abc", "module_abc", "module_xyz"};
+const std::array<std::string, kNumModules> kFilePaths{
+    "/usr/subpath/to/module_abc", "/local/subpath/to/module_abc", "/usr/subpath/to/module_xyz"};
+const std::array<std::string, kNumModules> kBuildIds{"build_id_0", "build_id_1", "build_id_2"};
+
+// ModulesDataView also has column index constants defined, but they are declared as private.
+constexpr int kColumnName = 0;
+constexpr int kColumnPath = 1;
+constexpr int kColumnAddressRange = 2;
+constexpr int kColumnFileSize = 3;
+constexpr int kColumnLoaded = 4;
+constexpr int kNumColumns = 5;
+
+std::string GetExpectedDisplayAddressRangeByIndex(size_t index) {
+  return absl::StrFormat("[%016" PRIx64 " - %016" PRIx64 "]", kStartAddresses[index],
+                         kEndAddresses[index]);
+}
+
+std::string GetExpectedDisplayFileSizeByIndex(size_t index) {
+  return orbit_display_formats::GetDisplaySize(kSizes[index]);
+}
+
+class ModulesDataViewTest : public testing::Test {
+ public:
+  explicit ModulesDataViewTest() : view_{&app_} {
+    for (size_t i = 0; i < kNumModules; i++) {
+      ModuleInMemory module_in_memory{kStartAddresses[i], kEndAddresses[i], kFilePaths[i],
+                                      kBuildIds[i]};
+      modules_in_memory_.push_back(module_in_memory);
+
+      ModuleInfo module_info{};
+      module_info.set_name(kNames[i]);
+      module_info.set_file_path(kFilePaths[i]);
+      module_info.set_build_id(kBuildIds[i]);
+      module_info.set_file_size(kSizes[i]);
+      EXPECT_TRUE(module_manager_.AddOrUpdateModules({module_info}).empty());
+    }
+  }
+
+  void AddModulesByIndices(const std::vector<size_t>& indices) {
+    std::set index_set(indices.begin(), indices.end());
```

Comment:
Why the set? Because of sorting? Where is that needed?

Reply:
Just in case that we input repeated indices 🙈  Of course we will not do it...